c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1525 @@
+    //! The eight individual routes making up a P2P table entry
+    struct DOXYNAME(routes) {
+        router_p2p_route route1 : 3;    //!< First packed route
+        router_p2p_route route2 : 3;    //!< Second packed route
+        router_p2p_route route3 : 3;    //!< Third packed route
+        router_p2p_route route4 : 3;    //!< Fourth packed route
+        router_p2p_route route5 : 3;    //!< Fifth packed route
+        router_p2p_route route6 : 3;    //!< Sixth packed route
+        router_p2p_route route7 : 3;    //!< Seventh packed route
+        router_p2p_route route8 : 3;    //!< Eighth packed route
+    };
+    uint value;                         //!< Overall entry packed as number
+} router_p2p_table_entry_t;
+
+ASSERT_WORD_SIZED(router_control_t);
+ASSERT_WORD_SIZED(router_packet_header_t);
+ASSERT_WORD_SIZED(router_error_status_t);
+ASSERT_WORD_SIZED(router_dump_outputs_t);
+ASSERT_WORD_SIZED(router_dump_status_t);
+ASSERT_WORD_SIZED(router_diagnostic_counter_ctrl_t);
+ASSERT_WORD_SIZED(router_timing_counter_ctrl_t);
+ASSERT_WORD_SIZED(router_diversion_t);
+ASSERT_WORD_SIZED(router_fixed_route_routing_t);
+ASSERT_WORD_SIZED(router_diagnostic_filter_t);
+ASSERT_WORD_SIZED(router_multicast_route_t);
+ASSERT_WORD_SIZED(router_p2p_table_entry_t);
+
+//! Router controller registers
+static volatile router_t *const router_control = (router_t *) RTR_BASE;
+//! Router diagnostic filters
+static volatile router_diagnostic_filter_t *const router_diagnostic_filter =
+        (router_diagnostic_filter_t *) (RTR_BASE + 0x200);
+//! Router diagnostic counters
+static volatile uint *const router_diagnostic_counter =
+        (uint *) (RTR_BASE + 0x300);
+//! Router multicast route table
+static volatile router_multicast_route_t *const router_multicast_table =
+        (router_multicast_route_t *) RTR_MCRAM_BASE;
+//! Router multicast key table (write only!)
+static volatile uint *const router_key_table = (uint *) RTR_MCKEY_BASE;
+//! Router multicast mask table (write only!)
+static volatile uint *const router_mask_table = (uint *) RTR_MCMASK_BASE;
+//! Router peer-to-peer route table
+static volatile router_p2p_table_entry_t *const router_p2p_route_table =
+        (router_p2p_table_entry_t *) RTR_P2P_BASE;
+
+//! \}
+
+// ---------------------------------------------------------------------
+// 11. Inter-chip transmit and receive interfaces
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 12. System NoC
+
+// No registers
+
+// ---------------------------------------------------------------------
+//! \name 13. SDRAM interface
+//! \brief The SDRAM interface connects the System NoC to an off-chip SDRAM
+//!     device. It is the ARM PL340, described in ARM document DDI 0331D.
+//! \details Only meaningfully usable by the monitor core when initialising the
+//!     overall chip. Use at other times is very much not recommended.
+//! \warning Do not use these without talking to Luis first!
```

Comment:
what happens if luis no longer works here? also, or what? ya going to steal my bacon supply?

Reply:
Or else we'll accuse you of Dave-level low level hackery. You have been warned!

Reply:
this does not seem to be a costly threat. my point was that the comment should explain WHY to talk to luis first.

Reply:
The point is I know that Luis will tell anyone thinking of messing with that to not do it. 😉

Reply:
fine, not caring enough to make it more clear.